hapi/hedera-protobufs/platform/event/tss_message_transaction.proto

```diff
@@ -0,0 +1,64 @@
+/**
+ * # Tss Message Transaction
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.event.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The TssMessageTransaction is used to send a TssMessage to the network for a candidate roster.
+ */
+message TssMessageTransaction {
```

Comment:
I think this and the `TssVoteTransaction` should be messages ending with `TransactionBody` that live in the same folder with the other `*TransactionBody` messages (e.g. `CryptoDeleteTransactionBody` [here](https://github.com/hashgraph/hedera-services/blob/develop/hapi/hedera-protobufs/services/crypto_delete.proto#L36)).

And then we should include these in the `TransactionBody#data` oneof [here](https://github.com/hashgraph/hedera-services/blob/develop/hapi/hedera-protobufs/services/transaction_body.proto#L133).

Reply:
Have updated. Thanks @tinker-michaelj 👍

Reply:
Ah tyvm @thomas-swirlds-labs ...I had only two other comments:
  1. We need corresponding `HederaFunctionality` enum values [here](https://github.com/hashgraph/hedera-services/blob/develop/hapi/hedera-protobufs/services/basic_types.proto#L1251).
  2. Since TSS is now an application service, would it make more sense to move its state definitions (tss_{vote,message}_map_key.proto) files under a _services/state/tss_ directory?